hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -698,11 +699,12 @@ public void loadAll(Set<K> keys, boolean replaceExistingValues) {
         MapKeySetRequest request = new MapKeySetRequest(name);
         MapKeySet mapKeySet = invoke(request);
         Set<Data> keySetData = mapKeySet.getKeySet();
-        Set<K> keySet = new HashSet<K>(keySetData.size());
+        InflatableSet<K> keySet = new InflatableSet<K>(keySetData.size());
         for (Data data : keySetData) {
             final K key = toObject(data);
```

Comment:
Sure. However that's yet another responsibility. There are no serialized entries to begin with. We would need some sort of lazy-generator.